### PR TITLE
Various typos fixes

### PR DESCRIPTION
**Pull request by marcwrobel**
This PR fixes various typos in :
- spring-boot-cli and spring-boot-actuator-autoconfigure javadoc,
- spring-boot-autoconfigure, spring-boot-actuator, spring-boot-tools, spring-boot-smoke-tests javadoc / tests,
- spring-boot tests, javadoc and code (the private method `remainderIsNotAlphaNumberic`'s name has been fixed),
- actuator documentation,
- `Metadata.MetadataItemCondition#findItem` `IllegalStateException` message,
- `spring.liquibase.check-change-log-location` deprecation reason,
- `PrometheusPushGatewayManager` log message.

The last three commits may be problematic and need special attention.

Multiple commits has been made. Let me know if you need multiple PRs.

### CODE REVIEW

File: spring-boot-project/spring-boot-cli/src/main/java/org/springframework/boot/cli/command/init/ProjectGenerationRequest.java [hunk 1]
```diff
@@ -191,7 +191,7 @@ void setPackageName(String packageName) {
 	}
 
 	/**
-	 * The type of project to generate. Should match one of the advertized type that the
+	 * The type of project to generate. Should match one of the advertised type that the
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-cli/src/main/java/org/springframework/boot/cli/command/init/ProjectGenerationRequest.java:194]: TIL that US English doesn't use a z here.

File: spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONArray.java [hunk 2]
```diff
@@ -39,7 +39,7 @@
  * In particular, {@code get} fails if the requested index holds the null reference, but
  * succeeds if it holds {@code JSONObject.NULL}.
  * <p>
- * Instances of this class are not thread safe. Although this class is nonfinal, it was
+ * Instances of this class are not thread safe. Although this class is non-final, it was
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONArray.java:42]: This class is "shaded" from third-party source and should be left as-is.

File: spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONObject.java [hunk 3]
```diff
@@ -708,7 +708,7 @@ public String toString() {
 	}
 
 	/**
-	 * Encodes this object as a human readable JSON string for debugging, such as: <pre>
+	 * Encodes this object as a human-readable JSON string for debugging, such as: <pre>
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONObject.java:711]: This class is "shaded" from third-party source and should be left as-is.

File: spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONStringer.java [hunk 4]
```diff
@@ -50,7 +50,7 @@
  * create more than 20 levels of nesting may fail with a {@link JSONException}.
  * <p>
  * Each stringer may be used to encode a single top level value. Instances of this class
- * are not thread safe. Although this class is nonfinal, it was not designed for
+ * are not thread safe. Although this class is non-final, it was not designed for
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONStringer.java:53]: This class is "shaded" from third-party source and should be left as-is.

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/source/ConfigurationPropertyName.java [hunk 5]
```diff
@@ -448,7 +448,7 @@ private boolean defaultElementEquals(Elements e1, Elements e2, int i) {
 		int i2 = 0;
 		while (i1 < l1) {
 			if (i2 >= l2) {
-				return remainderIsNotAlphaNumberic(e1, i, i1);
+				return remainderIsNotAlphaNumeric(e1, i, i1);
```
Comment (wilkinsona) [inline spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/source/ConfigurationPropertyName.java:451]: For consistency with `alpha-numeric -> alphanumeric`, this method should be renamed to `remainderIsNotAlphanumeric`.

**Review by wilkinsona — CHANGES_REQUESTED** [review]
Thanks very much for the PR. I've left a few comments for your consideration when you have time.

File: spring-boot-project/spring-boot-cli/src/main/java/org/springframework/boot/cli/command/init/ProjectGenerationRequest.java (hunk 1 above)
Comment (marcwrobel) [reply spring-boot-project/spring-boot-cli/src/main/java/org/springframework/boot/cli/command/init/ProjectGenerationRequest.java:194]: ;)

File: spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/source/ConfigurationPropertyName.java (hunk 5 above)
Comment (marcwrobel) [reply spring-boot-project/spring-boot/src/main/java/org/springframework/boot/context/properties/source/ConfigurationPropertyName.java:451]: Done.

File: spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONArray.java (hunk 2 above)
Comment (marcwrobel) [reply spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONArray.java:42]: Done.

File: spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONObject.java (hunk 3 above)
Comment (marcwrobel) [reply spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONObject.java:711]: Done.

File: spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONStringer.java (hunk 4 above)
Comment (marcwrobel) [reply spring-boot-project/spring-boot-tools/spring-boot-configuration-processor/src/json-shade/java/org/springframework/boot/configurationprocessor/json/JSONStringer.java:53]: Done.

File: spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jackson/JacksonProperties.java [hunk 6]
```diff
@@ -56,8 +56,8 @@ public class JacksonProperties {
 	private String propertyNamingStrategy;
 
 	/**
-	 * Jackson visibility thresholds that can be used to limit which methods (and fields)
-	 * are auto-detected.
+	 * Jackson's visibility thresholds that can be used to limit which methods (and
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/jackson/JacksonProperties.java:59]: Done.

File: spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/datadog/DatadogProperties.java [hunk 7]
```diff
@@ -36,8 +36,8 @@ public class DatadogProperties extends StepRegistryProperties {
 	private String apiKey;
 
 	/**
-	 * Datadog application key. Not strictly required, but improves the Datadog experience
-	 * by sending meter descriptions, types, and base units to Datadog.
+	 * Datadog's application key. Not strictly required, but improves the Datadog
```
[the remark this answers is not in the document]
Comment (marcwrobel) [reply spring-boot-project/spring-boot-actuator-autoconfigure/src/main/java/org/springframework/boot/actuate/autoconfigure/metrics/export/datadog/DatadogProperties.java:39]: Done.

**Comment by marcwrobel** [comment]
> Thanks very much for the PR. I've left a few comments for your consideration when you have time.

Thanks for this review. Requested changes has been done.

**Comment by marcwrobel** [comment]
@wilkinsona, looks good to me.

**Pull request closed without merging**